pkg/cache/groupcache_test.go

```diff
@@ -16,32 +17,43 @@ import (
 	"github.com/go-kit/log"
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/common/route"
+	"github.com/thanos-io/thanos/pkg/component"
 	"github.com/thanos-io/thanos/pkg/discovery/dns"
 	"github.com/thanos-io/thanos/pkg/model"
 	"github.com/thanos-io/thanos/pkg/objstore"
+	"github.com/thanos-io/thanos/pkg/prober"
+	httpserver "github.com/thanos-io/thanos/pkg/server/http"
 	"github.com/thanos-io/thanos/pkg/store/cache/cachekey"
 	"github.com/thanos-io/thanos/pkg/testutil"
+	galaxyhttp "github.com/vimeo/galaxycache/http"
+	"golang.org/x/net/http2"
 )
 
-// Benchmark retrieval of one key from one groupcache node.
-func BenchmarkGroupcacheRetrieval(b *testing.B) {
+const basePath = `/_groupcache/`
+const groupName = `groupName`
+const selfURLH1 = "http://localhost:12345"
+const selfURLH2C = "http://localhost:12346"
+
+func TestMain(m *testing.M) {
 	reg := prometheus.NewRegistry()
 	router := route.New()
 
 	bkt := objstore.NewInMemBucket()
-	b.Cleanup(func() { bkt.Close() })
+	defer bkt.Close()
```

Comment:
```suggestion
	b.Cleanup(func() { bkt.Close() })
```
Why do we go back to defer here? AFAIK there's no functional difference and `Cleanup` better to avoid errors.

Reply:
It was a benchmark function before, now it is `TestMain(m *testing.M)`. The `M` type doesn't have those methods: https://pkg.go.dev/testing#M. Thus, `defer` is used because there's no better option. I hope that explains this change :smile: